cmd/kcp/kcp.go

```diff
@@ -219,7 +223,37 @@ func main() {
 				prepared := server.PrepareRun()
 
 				return prepared.Run(ctx.Done())
-			})
+			}
+
+			if len(etcdClientInfo.Endpoints) == 0 {
+				// No etcd servers specified so create one in-process:
+				return s.Run(runFunc)
+			} else {
+				etcdClientInfo.TLS = &tls.Config{
+					InsecureSkipVerify: true,
+				}
+
+				if len(etcdClientInfo.CertFile) > 0 && len(etcdClientInfo.KeyFile) > 0 {
+					cert, err := tls.LoadX509KeyPair(etcdClientInfo.CertFile, etcdClientInfo.KeyFile)
+					if err != nil {
+						return fmt.Errorf("failed to load x509 keypair: %s", err)
+					}
+					etcdClientInfo.TLS.Certificates = []tls.Certificate{cert}
+				}
+
+				if len(etcdClientInfo.TrustedCAFile) > 0 {
+					if caCert, err := ioutil.ReadFile(etcdClientInfo.TrustedCAFile); err != nil {
+						return fmt.Errorf("failed to read ca file: %s", err)
+					} else {
+						caPool := x509.NewCertPool()
+						caPool.AppendCertsFromPEM(caCert)
+						etcdClientInfo.TLS.RootCAs = caPool
```

Comment:
At this point, wouldn't you want to set `etcdClientInfo.TLS.InsecureSkipVerify = false`?

Reply:
Good catch, I had that in there at one point and deleted it somehow. Thanks!